### PR TITLE
Fixes double call to view_will_appear

### PR DESCRIPTION
**Pull request by squidpunch**

### CODE REVIEW

File: lib/project/ext/ui_view_controller.rb [hunk 1]
```diff
@@ -79,14 +79,16 @@ def viewDidLoad
...
+  def pm_handles_delegates?
     self.is_a?(ProMotion::ViewController) || self.is_a?(ProMotion::TableScreen)
   end
 
   def view_will_appear(animated)
   end
   def viewWillAppear(animated)
-    self.view_will_appear(animated)
+    unless pm_handles_delegates?
+      self.view_will_appear(animated)
```
Comment (squidpunch) [inline lib/project/ext/ui_view_controller.rb]: similar to viewDidLoad - this is called in PM, so we don't want to call it a second time around.

**Comment by markrickert** [comment]
Tests? :)

**Comment by squidpunch** [comment]
yeah I got lazy since I was removing something and depending on PM base - I'll see what i can whip up as far as testing it @markrickert

**Comment by squidpunch** [comment]
now with tests @markrickert !

**Comment by jamonholmgren** [comment]
:+1:

**Comment by GantMan** [comment]
dont try to skip tests with the rickert around :)

**Comment by jamonholmgren** [comment]
> dont try to skip tests with the rickert around :)

He keeps us honest. haha

**Comment by squidpunch** [comment]
heh, well if you look at my contributions to RMQ you'll see I am a pretty big stickler for testing too - I just didnt even think to test something I was removing, but glad it came up.  I started another branch to restructure the tests a bit (kind of like how Promotion has test screens instead of defining them in the spec) and make sure the files match up (stuff we moved to ext making sure tests are in matching spec files etc) - going to PR that sometime after this is merged :+1:

**Comment by GantMan** [comment]
hah, it's all in good jest squid!  Tests everywhere :) !

O btw, if anyone thinks of a good way to mix benchmarking with tests, let me know.  I think it would be good to have tests fail if someone makes a change that significantly slows down a core function.  

Ideas abound!

**Comment by jamonholmgren** [comment]
I think it's a good time to mention that @Watson1978's https://github.com/Watson1978/motion-benchmark-ips is an *excellent* benchmarking tool for RubyMotion.

**Comment by GantMan** [comment]
agreed!  I've used it often.  Ideas on how to integrate that into tests?

**Comment by jamonholmgren** [comment]
Maybe Watson could help us. :)

**Comment by markrickert** [comment]
I just integrated this branch into my current project and I'm seeing the problem is fixed!

**Comment by squidpunch** [comment]
great to hear @markrickert  !

**Comment by squidpunch** [comment]
ok sounds like this fixes the issue I'm pulling a @GantMan and merging this myself :laughing:

**Comment by GantMan** [comment]
hahahahaha.   Sometimes you gotta just hit merge brotha!

**Pull request merged**